plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/issues/RejectOpenDocumentsIssuePublisher.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2020 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.xtext.server.issues;
+
+import java.util.List;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.n4js.ide.xtext.server.ResourceTaskManager;
+import org.eclipse.n4js.ide.xtext.server.build.XBuildRequest.AfterValidateListener;
+import org.eclipse.n4js.xtext.server.LSPIssue;
+import org.eclipse.xtext.validation.Issue;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * Implementation of an AfterValidateListener that will forward issues to a a delegate if the underlying document is not
+ * open.
+ */
+@Singleton
+@SuppressWarnings("deprecation")
+public class RejectOpenDocumentsIssuePublisher implements AfterValidateListener {
```

Comment:
The fields for `IssueAcceptor`s are often called "issuePublisher" or "publisher". So to keep listeners and acceptors separate better use suffix "...Listener" in the name of this class.

Reply:
Done